HLTrigger/HLTfilters/plugins/L1TTkMuonFilter.cc

```diff
@@ -26,6 +27,31 @@
 // constructors and destructor
 //
 
+namespace {
+  bool passMuQual(const l1t::TkMuon& muon) {
+    if (muon.muonDetector() == 3) {
+      int quality = muon.quality();
+      if (quality == 11 || quality == 13 || quality == 14 || quality == 15) {
+        return true;
+      } else {
+        return false;
+      }
+    } else {
+      return true;
+    }
+  }
+  bool isDupMuon(const l1t::TkMuonRef& muon, const std::vector<l1t::TkMuonRef>& existing) {
+    for (const auto& exist : existing) {
+      //it is our understanding that there is an exact eta phi match
+      //and we should not be concerned with numerical precision
+      if (reco::deltaR2(*muon, *exist) <= 0) {
```

Comment:
are the duplicate muons always from the same subsystem (barrel/overlap.forward) ?
or are the coordinates exactly the same also for muons from different subsystems ?

Reply:
I dont know.  I do know the cut listed on the twiki is deltaR>0